perfkitbenchmarker/linux_packages/maven.py

```diff
@@ -70,22 +96,48 @@ def _Install(vm):
   maven_full_ver = FLAGS.maven_version
   maven_major_ver = maven_full_ver[:maven_full_ver.index('.')]
   maven_url = MVN_URL.format(maven_major_ver, maven_full_ver)
-  vm.RemoteCommand(
-      ('mkdir {0} && curl -L {1} | '
-       'tar -C {0} --strip-components=1 -xzf -').format(MVN_DIR, maven_url))
-
-  # Get JAVA_HOME
-  out, _ = vm.RemoteCommand('readlink -f `which java`')
-  out = out.strip()
-  java_home = out[:out.index('/jre')]
+  maven_tar = maven_url.split('/')[-1]
+  if maven_tar not in PREPROVISIONED_DATA:
+    PREPROVISIONED_DATA[maven_tar] = ''  # will only work with preprovision_ignore_checksum
+    PACKAGE_DATA_URL[maven_tar] = maven_url
+  maven_remote_path = posixpath.join(INSTALL_DIR, maven_tar)
+  vm.InstallPreprovisionedPackageData(
+      PACKAGE_NAME,
+      [maven_tar],
+      INSTALL_DIR
+  )
+  vm.RemoteCommand(('mkdir -p {0} && '
+                    'tar -C {0} --strip-components=1 -xzf {1}').format(
+                        MVN_DIR, maven_remote_path))
+
+  java_home = _GetJavaHome(vm)
 
   # Set env variables for maven
   maven_env = MVN_ENV.format(java_home=java_home, maven_home=MVN_DIR)
   cmd = 'echo "{0}" | sudo tee -a {1}'.format(maven_env, MVN_ENV_PATH)
   vm.RemoteCommand(cmd)
 
+  # On CentOS, the mvn command looks for tools.jar in $JAVA_HOME/../lib
```

Comment:
This seems a little inelegant, and in my testing unnecessary.
 
On A CentOS 8 VM mvn 3.6.3 found the tools jar of OpenJDK 8 regardless of whether JAVA_HOME was unset, set to /usr/lib/jvm/java/ or /usr/lib/jvm/java/jre

Reply:
@pkmc this is something I experienced on a particular CentOS 7 static VM, with maven 3.6.1. I haven't been able to reproduce it on any cloud provider.

Reply:
I have removed this, we can revisit in the future if anyone has issues with the lib path.